### PR TITLE
Me: Update Account to use Redux analytics

### PR DESCRIPTION
**Pull request by tyxla**
This PR refactors `AccountPassword` to:

* Use Redux analytics instead of `eventRecorder`.
* Remove unnecessary `bindActionCreators` usage.

This PR should not introduce any visual or behavioral changes.

Part of #20053.

To test:
* Checkout this branch
* Make sure your ga debug is enabled: `localStorage.debug = 'calypso:analytics:ga,calypso:analytics:utils'`
* Go to http://calypso.localhost:3000/me/account
* Verify the right event is being dispatched when checking the Community Translator checkbox (you need to select a different interface language than English to reveal this setting).
* Verify the right event is being dispatched when clicking the Community Translator "Learn More" link (you need to select a different interface language than English to reveal this setting).
* Verify the right event is being dispatched when checking the Holiday Snow checkbox (you need to set your current system date to any day in the December month to reveal this setting).
* Verify the right event is being dispatched when clicking the "Add New Site" link in the "Primary Site" field (you need to have no sites to reveal this setting).
* Verify the right event is being dispatched when focusing the "Email Address" field.
* Verify the right event is being dispatched when focusing the "Web Address" field.
* Verify the right event is being dispatched when clicking the "Interface Language" field.
* Verify the right event is being dispatched when ticking a radio button in the "Would you like a matching blog address too?" field (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
* Verify the right event is being dispatched when focusing the "Confirm Username" field (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
* Verify the right event is being dispatched when clicking the "My Profile" link in the "Please Read Carefully" section (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
* Verify the right event is being dispatched when clicking the "Save Username" button (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
* Verify the right event is being dispatched when clicking the "Cancel" button (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
* Verify the right event is being dispatched when focusing the username field (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
* Verify there are no regressions in the account page by testing as a11n and non-a11n.

### CODE REVIEW

**Comment by sirreal** [comment]
- ✅Verify the right event is being dispatched when checking the Community Translator checkbox (you need to select a different interface language than English to reveal this setting).
- ❗️Verify the right event is being dispatched when clicking the Community Translator "Learn More" link (you need to select a different interface language than English to reveal this setting).
- ✅ Verify the right event is being dispatched when checking the Holiday Snow checkbox (you need to set your current system date to any day in the December month to reveal this setting).
- ✅ Verify the right event is being dispatched when clicking the "Add New Site" link in the "Primary Site" field (you need to have no sites to reveal this setting).
- ✅ Verify the right event is being dispatched when focusing the "Email Address" field.
- ✅ Verify the right event is being dispatched when focusing the "Web Address" field.
- ✅ Verify the right event is being dispatched when clicking the "Interface Language" field.
- ✅ Verify the right event is being dispatched when ticking a radio button in the "Would you like a matching blog address too?" field (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
- ✅ Verify the right event is being dispatched when focusing the "Confirm Username" field (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
- ✅ Verify the right event is being dispatched when clicking the "My Profile" link in the "Please Read Carefully" section (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
- ✅ Verify the right event is being dispatched when clicking the "Save Username" button (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
- ✅ Verify the right event is being dispatched when clicking the "Cancel" button (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
- ✅ Verify the right event is being dispatched when focusing the username field (you need to be using a non-a11n account and input a different username in the username field to reveal this setting).
- ✅ Verify there are no regressions in the account page by testing as a11n and non-a11n.
  - I smoke tested a reasonable amount of functionality 🙂 


Note:

Related to the way analytics is called —via `eventRecorder` directly invoking lib/analytics on master or via analytics middleware in the case of this branch— the debug output is slighly different, but this appears to be a simple side effect of the middleware implementation and nothing that impacts the behavior of the analytics code:

- **master** `calypso:analytics:ga [Disallowed] analytics recordEvent( (2) ["Me", "Clicked on Community Translator Learn More Link"] )`
- **branch** `calypso:analytics:ga [Disallowed] analytics recordEvent( (4) ["Me", "Clicked on Community Translator Learn More Link", undefined, undefined] )`

File: client/me/account/main.jsx [hunk 1]
```diff
@@ -257,6 +255,40 @@ const Account = createReactClass( {
...
+	getCheckboxHandler( checkboxName ) {
+		return event => {
+			const action = 'Clicked ' + checkboxName + ' checkbox';
+			const value = event.target.checked ? 1 : 0;
+
+			this.props.recordGoogleEvent( 'Me', action, 'checked', value );
+		};
+	},
+
+	getRadioHandler( radioName ) {
```
Comment (sirreal) [inline client/me/account/main.jsx]: This is only used to generate 1 handler. Could we just make a dedicated method to track that event?

Comment (tyxla) [reply client/me/account/main.jsx]: 👍 makes sense - done in https://github.com/Automattic/wp-calypso/pull/22881/commits/e9bef140c2e2346bd1590bc16d4a3957e0cbefce

File: client/me/account/main.jsx [hunk 2]
```diff
@@ -760,7 +792,7 @@ export default compose(
 		state => ( {
 			requestingMissingSites: isRequestingMissingSites( state ),
 		} ),
-		dispatch => bindActionCreators( { successNotice, errorNotice, recordTracksEvent }, dispatch )
+		{ successNotice, errorNotice, recordGoogleEvent, recordTracksEvent }
```
[the remark this answers is not in the document]
Comment (tyxla) [reply client/me/account/main.jsx]: 👍 done in https://github.com/Automattic/wp-calypso/pull/22881/commits/0f38ade88063af8ee018fc182af47df2bbb8a707

**Comment by sirreal** [comment]
🚢 !

**Pull request merged**